### ISSUE TITLE
Return more explicit error when config file can't be found

### ISSUE THREAD

**Issue by tschuy** [spf13/viper#210]
Right now if viper can't find a config file in any of the config paths, `viper.ReadInConfig()` returns `Unsupported Config Type ""`. Instead, it'd be nice to get an error that's a little more explicit, like the `ConfigFileNotFoundError` currently being [swallowed](https://github.com/spf13/viper/blob/master/viper.go#L1246) (and then [swallowed again](https://github.com/spf13/viper/blob/master/viper.go#L1234)).

**Comment by Waitak** [comment]
This happens because ReadInConfig() calls getConfigType(), which calls getConfigFile() before subsequently calling getConfigFile() itself:

```.go
func (v *Viper) ReadInConfig() error {
	jww.INFO.Println("Attempting to read in config file")
	if !stringInSlice(v.getConfigType(), SupportedExts) {
		return UnsupportedConfigError(v.getConfigType())
	}

	file, err := afero.ReadFile(v.fs, v.getConfigFile())
```

One solution would be to explicitly call findConfigFile() first. This also avoids calling it twice, and gives the correct error if it fails.

```.go
func (v *Viper) ReadInConfig() error {
	jww.INFO.Println("Attempting to read in config file")

	cf, err := v.findConfigFile()
	if err != nil {
		return err
	}
	v.configFile = cf

	if !stringInSlice(v.getConfigType(), SupportedExts) {
		return UnsupportedConfigError(v.getConfigType())
	}

	file, err := afero.ReadFile(v.fs, v.getConfigFile())
```

**Comment by awishformore** [comment]
Merged as part of #161
